config/cmake/HDFCompilerFlags.cmake

```diff
@@ -211,12 +225,16 @@ if (HDF5_ENABLE_DEV_WARNINGS)
       list (APPEND H5_CFLAGS "-Winline")
     elseif (CMAKE_C_COMPILER_ID STREQUAL "Intel" AND NOT _INTEL_WINDOWS)
       list (APPEND H5_CFLAGS "-Winline")
+    elseif (CMAKE_C_COMPILER_ID MATCHES "IntelLLVM" AND NOT _INTEL_WINDOWS)
+      list (APPEND H5_CFLAGS "-Winline")
+    elseif (CMAKE_C_COMPILER_ID MATCHES "[Cc]lang")
+      list (APPEND H5_CFLAGS "-Winline")
     endif ()
   endif ()
 else ()
-  if (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 4.8)
```

Comment:
Is the GNU "no-developer-general" section not needed here anymore?

Reply:
whoops - might be too aggressive delete!

Reply:
Fixed